### PR TITLE
WIP :  adding support for default bucketpolicy in OBC

### PR DESCRIPTION
**Pull request by thotz**
**Description of your changes:**
adding support for default bucketpolicy in OBC
**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by galexrt** [comment]
@thotz Ping. The CI failure seems to be a known issue. Please rebase on latest `master` branch to see if that fixes the CI issue.

Are you available to continue working on this PR?

**Comment by thotz** [comment]
Adding comment make it live

File: cluster/examples/kubernetes/ceph/storageclass-bucket-retain.yaml [hunk 1]
```diff
@@ -9,6 +9,18 @@ parameters:
   objectStoreName: my-store # port 80 assumed
   objectStoreNamespace: rook-ceph
   region: us-east-1
+  bucketpolicy: '{
```
Comment (BlaineEXE) [inline cluster/examples/kubernetes/ceph/storageclass-bucket-retain.yaml:12]: What are the effects of this bucket policy?

**Comment by thotz** [comment]
I am closing this PR since no longer planning to work on this feature. I will send a PR to remove the design doc from Rook code repo

**Pull request closed without merging**